diesel/src/migrations/schema.rs

```diff
@@ -7,10 +7,53 @@ table! {
 
 #[derive(Debug, Copy, Clone)]
 pub struct NewMigration<'a>(pub &'a str);
-impl_Insertable! {
-    (__diesel_schema_migrations)
-    pub struct NewMigration<'a>(
-        #[column_name(version)]
-        pub &'a str,
-    );
+
+use insertable::{Insertable, InsertValues, ColumnInsertValue};
+use backend::Backend;
+use query_builder::insert_statement::{IntoInsertStatement, InsertStatement};
+use expression::helper_types::AsNullableExpr;
+
+impl<'insert, 'a, DB> Insertable<self::__diesel_schema_migrations::table, DB>
+    for &'insert NewMigration<'a>
+    where DB: Backend,
+               (ColumnInsertValue<self::__diesel_schema_migrations::version,
+                   AsNullableExpr<&'insert &'a str,
+                       self::__diesel_schema_migrations::version>>, ): InsertValues<DB>,
+{
+    type Values = (ColumnInsertValue<self::__diesel_schema_migrations::version,
+                   AsNullableExpr<&'insert &'a str,
+                                  self::__diesel_schema_migrations::version>>, );
+
+    fn values(self) -> Self::Values {
+        use expression::{AsExpression, Expression};
+        use insertable::ColumnInsertValue;
+        use types::IntoNullable;
+        let ref version = self.0;
+        (ColumnInsertValue::Expression(self::__diesel_schema_migrations::version,
+            AsExpression::<<<self::__diesel_schema_migrations::version
+                as Expression>::SqlType as IntoNullable>::Nullable>
+            ::as_expression(version)),)
```

Comment:
Can you try not to break this at `as` and `::`? It's really hard to read this way. Using the `Table` alias from abovem you could write

```rust
(ColumnInsertValue::Expression(Version,
    AsExpression::<
        <<Version as Expression>::SqlType as IntoNullable>::Nullable
    >::as_expression(version)
),)
```